File: src/main/java/com/dafttech/terra/engine/gui/anchors/AnchorCenterX.java
package com.dafttech.terra.engine.gui.anchors;

import com.dafttech.terra.engine.gui.GUIObject;
import com.dafttech.terra.engine.gui.containers.GUIContainer;

public class AnchorCenterX extends GUIAnchor {
    @Override
    public void applyAnchor(GUIObject object, GUIContainer container) {
        object.position.x = container.size.x / 2 - object.size.x / 2;
    }
}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/anchors/AnchorCenterX.scala
package com.dafttech.terra.engine.gui.anchors

import com.dafttech.terra.engine.gui.GUIObject
import com.dafttech.terra.engine.gui.containers.GUIContainer

class AnchorCenterX extends GUIAnchor {
  def applyAnchor(gObj: GUIObject, container: GUIContainer) {
    gObj.position = gObj.position.withX(container.size.x / 2 - gObj.size.x / 2)
  }
}